apps/rush/src/actions/BaseAction.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import {
+  CommandLineAction,
+  ICommandLineActionOptions
+} from '@microsoft/ts-command-line';
+
+import {
+  RushConfiguration
+} from '@microsoft/rush-lib';
+
+import EventHooksManager from '../utilities/EventHooksManager';
+
+export abstract class BaseAction extends CommandLineAction {
+  private _rushConfiguration: RushConfiguration;
+  private _eventHooksManager: EventHooksManager;
+
+  constructor(options: ICommandLineActionOptions) {
+    super(options);
+  }
+
+  protected onExecute(): void {
+    this._ensureEnvironment();
+    this.run();
+  }
+
+  protected abstract run(): void;
+
+  protected get rushConfiguration(): RushConfiguration {
+    if (!this._rushConfiguration) {
+      this._rushConfiguration = RushConfiguration.loadFromDefaultLocation();
```

Comment:
>this._rushConfiguration = RushConfiguration.loadFromDefaultLocation(); [](start = 6, length = 70)

This throws exceptions as part of its normal expected operation, i.e. when we need to tell the end user about mistakes in their rush.json file.  I feel like that kind of thing should happen during a well-defined step of the algorithm, versus magically occurring whenever someone happens to look at the variable.

Certain commands do not need the configuration file (e.g. "rush scan" and eventually "rush init").  So I'd propose a design like this:

The base class normally calls this.loadConfigFile() before run().  However, a subclass can suppress this by passing an option to the BaseAction constructor.